### PR TITLE
Create Progress indicator (#267)

### PR DESCRIPTION
**Pull request by khashayar-pakkhesal**

### CODE REVIEW

**Comment by khashayar-pakkhesal** [comment]
this pr closes #267

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/ProgressIndicator/BitProgressIndicator.razor.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+
+namespace Bit.Client.Web.BlazorUI
+{
+    public partial class BitProgressIndicator
+    {
+        private double percentComplete;
+        private bool HasSetPercent;
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/ProgressIndicator/BitProgressIndicator.razor.cs:13]: please change this variable name to "PercentCompleteHasBeenSet"

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/ProgressIndicator/BitProgressIndicator.razor.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        protected override string RootElementClass => "bit-pi";
+        protected override void RegisterComponentClasses()
+        {
+            ClassBuilder.Register(() => HasSetPercent ? string.Empty
+                                                : $"{RootElementClass}-indeterminate-{VisualClassRegistrar()}");
+        }
+
+        private static double Normalize(double value) => Math.Clamp(value, 0, 100);
+        private string ProgressTrackerWidth => HasSetPercent ? $"width: {percentComplete}%" : string.Empty;
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/ProgressIndicator/BitProgressIndicator.razor.cs:64]: based on a comment above, about moving this logic into the .razor file, this property is redundant and should be removed

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor [hunk 3]
```diff
@@ -798,5 +808,5 @@
 <div>
     <BitColorPicker @bind-Color="SelectedColor"></BitColorPicker>
 
-    @SelectedColor
+    @typeof(BitToggleButton)
```
Comment (msynk) [inline src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor:811]: what is this change?

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor.cs [hunk 4]
```diff
@@ -30,6 +31,28 @@ public partial class Component
         private double SpinButtonWithCustomHandlerValue = 14;
         private double SpinButtonWithLabelAboveValue = 7;
 
+        public int CompletedPercent { get; set; }
+        private string description = "Push button to start !";
+        private async Task HandleProgress()
```
Comment (msynk) [inline src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor.cs:36]: rename this method to "StartProgress"

**Review by msynk — CHANGES_REQUESTED** [review]

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/ProgressIndicator/BitProgressIndicator.razor [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+<div @ref="RootElement"
+     style="@StyleBuilder.Value"
+     class="@ClassBuilder.Value">
+    <div>
+        @Label
+    </div>
+    <div>
+        <div style="@($"height: {BarHeight}px;")"></div>
+        <div style="@($"{ProgressTrackerWidth}; height: {BarHeight}px;")"></div>
```
[the remark this answers is not in the document]
Comment (khashayar-pakkhesal) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/ProgressIndicator/BitProgressIndicator.razor]: Moved this logic to .razir

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/ProgressIndicator/BitProgressIndicator.razor.cs (hunk 1 above)
Comment (khashayar-pakkhesal) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/ProgressIndicator/BitProgressIndicator.razor.cs]: Renamed HasSetPercent to PercentCompleteHasBeenSet

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/ProgressIndicator/BitProgressIndicator.razor.cs (hunk 2 above)
Comment (khashayar-pakkhesal) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/ProgressIndicator/BitProgressIndicator.razor.cs]: Removed the property

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor (hunk 3 above)
Comment (khashayar-pakkhesal) [reply src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor]: Fully removed this line

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor.cs (hunk 4 above)
Comment (khashayar-pakkhesal) [reply src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor.cs]: Renamed HandleProgress to StartProgress

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/SpinButtons/BitSpinButton.razor [hunk 6]
```diff
@@ -15,13 +15,13 @@
                value="@($"{Value} {Suffix}")"
                @onchange="e => HandleOnInputChange(e)" />
         <span>
-            <button @onclick="e => HandleButtonClick(SpinButtonAction.Up)">
+            <button @onclick="e => HandleButtonClick(SpinButtonAction.Up)" class="@UpArrowButtonStyles">
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/SpinButtons/BitSpinButton.razor:18]: what is this change doing in this PR?

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/SpinButtons/BitSpinButton.razor [hunk 7]
```diff
@@ -15,13 +15,13 @@
...
                 <span>
                     <i class="bit-icon bit-icon--ChevronUpSmall bit-icon--s"></i>
                 </span>
             </button>
 
-            <button @onclick="e => HandleButtonClick(SpinButtonAction.Down)">
+            <button @onclick="e => HandleButtonClick(SpinButtonAction.Down)" class="@DownArrowButtonStyles">
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/SpinButtons/BitSpinButton.razor:24]: what is this change doing in this PR?

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/SpinButtons/BitSpinButton.razor.cs [hunk 8]
```diff
@@ -28,9 +28,19 @@ public double Value
                 _ = ValueChanged.InvokeAsync(value);
             }
         }
+        /// <summary>
+        /// Custom style for down arrow button
+        /// </summary>
+
+        [Parameter] public string? DownArrowButtonStyles { get; set; }
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/SpinButtons/BitSpinButton.razor.cs:35]: what is this change doing in this PR?

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/SpinButtons/BitSpinButton.razor.cs [hunk 9]
```diff
@@ -28,9 +28,19 @@ public double Value
...
 
         /// <summary>
-        /// Callback for when the spin button value change
+        /// Custom style for down arrow button
+        /// </summary>
+        [Parameter] public string? UpArrowButtonStyles { get; set; }
+
+        /// <summary>
+        /// Custom style for up arrow button
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/SpinButtons/BitSpinButton.razor.cs:43]: what is this change doing in this PR?

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor [hunk 10]
```diff
@@ -797,6 +807,4 @@
 
 <div>
     <BitColorPicker @bind-Color="SelectedColor"></BitColorPicker>
-
-    @SelectedColor
```
Comment (msynk) [inline src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor:801]: please do not apply changes unrelated to this PR

**Review by msynk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**